src/parser.rs

```diff
@@ -5288,80 +5295,46 @@ mod tests {
         });
     }
 
-    // TODO add tests for all data types? https://github.com/sqlparser-rs/sqlparser-rs/issues/2
-    // TODO when we have dialect validation by data type parsing, split test
-    #[test]
```

Comment:
seems like we have lost coverage after the rebase (e.g. the new tests for `TIME WITHOUT TIMEZONE` for example)

Reply:
@alamb that's expected, as I mentioned in my comment above.

I'll add tests along the week, trying to reach 100% coverage for data type parsing. But for now, I'd like to restrict this to the ones I added if possible

Reply:
I don't understand why this PR would *remove* coverage for types that were added in other PRs. I am fine with expanding coverage in future PRs

Reply:
Fair enough. It was because I (dumbly) did the structure for tests twice, as it didn't please me. I'll re-add the coverage for those data types, and for now leave as "additional data types" for those that aren't covered by ANSII